include/pybind11/cast.h

```diff
@@ -50,6 +50,18 @@ PYBIND11_NAMESPACE_BEGIN(detail)
 template <typename type, typename SFINAE = void> class type_caster : public type_caster_base<type> { };
 template <typename type> using make_caster = type_caster<intrinsic_t<type>>;
 
+template <typename T>
+struct is_generic_type<
```

Comment:
Two questions:

* Will this work if I add another specialization pair for `is_base_of<smart_holder_type_caster_base_tag, ...>`?

* What do you think about choosing a different name, to connect to the intended purpose instead of pegging the name on `type_caster_generic`, e.g. `is_iterable_t_compatible`? (Otherwise it will get really confusing that `generic` is connected to `smart_holder_type_caster`.)

Reply:
Done w/ db7ed8d - I removed `is_generic_type<>` as it's inclusion here may be a bit confusing

Will make a small separate PR